src/Microsoft.PowerShell.ConsoleHost/host/msh/ProgressNode.cs

```diff
@@ -387,14 +387,26 @@ internal static bool IsMinimalProgressRenderingEnabled()
                 maxWidth = PSStyle.Instance.Progress.MaxWidth;
             }
 
+            // if the activity is really long, only use up to half the width
+            string activity;
+            if (Activity.Length > maxWidth / 2)
+            {
+                activity = Activity.Substring(0, maxWidth / 2) + PSObjectHelper.Ellipsis;
```

Comment:
Looking on Archive expand example in #14414 I think we break the scenario.

Reply:
I just tried expand-archive with this branch and works fine with a narrow console window.  As expected, activity and status are truncated.

Reply:
I guess they are truncated too early while there is still a lot of free space on the right.

Reply:
The bar area is also used for Status text which can change as progress is being written.  So it's possible to fill up as much of Activity and Status as possible so whitespace is limited, it would also mean that the Status area would resize if the text changes which would make it hard to read.

Reply:
I tried the build and I see the resizing scenario doesn't work at all. But this doesn't depend on the PR.
Run follow and use mouse to change windows width:
```powershell
1..100 | % { Write-Progress -Activity "123456789012345678901234567890123456789012345678901234567890" -Status "S" -PercentComplete $_ -SecondsRemaining $_ ; Sleep 1}
```
Results:
- after size changed (both left and right directions) the progress bar doesn't work
- screen destroyed 
- after the command finished (or Ctrl-C pressed) screen buffer destroyed 
- instead of final Ellipsis I see single dot (direct output Ellipsis works on the console

(only last related to the PR)

Reply:
@iSazonov Can you please take another look and see if you have any outstanding concerns for the changes in this particular PR? For the points you brought up above in https://github.com/PowerShell/PowerShell/pull/15264#discussion_r618048122, they look like out of the scope of this PR (correct me if I'm wrong), and maybe track them in a separate issue?

Reply:
@daxian-dbw This removes the exception but I personally do not like the result of the display, although it is not fundamental. The fundamental thing is that the resizing scenario works horribly.  But I don't see the point of opening a new issue since MSFT team doesn't have the resources to fix 3000 issues anyway.

Reply:
I opened #15407 to track the resizing issue.

> instead of final Ellipsis I see single dot (direct output Ellipsis works on the console

![image](https://user-images.githubusercontent.com/127450/118310760-e041a700-b4a3-11eb-8771-8ab9ec3b4d61.png)

@SteveL-MSFT It does look weird to have a single dot instead of Ellipsis when truncation happens. Should that be addressed in this PR or you think it should be a separate one?

Reply:
I'll take a look at the other issues in a separate PR since this specific issue is affecting users with real world scenarios

Reply:
Sounds good. #15412 is opened to track the review comment.